Provider documentation fixes (#7823)
traefik/traefik: docs/content/providers/overview.md

Configuration discovery in Traefik is achieved through _Providers_.

The _providers_ are infrastructure components, whether orchestrators, container engines, cloud providers, or key-value stores.
The idea is that Traefik queries the provider APIs in order to find relevant information about routing,
and when Traefik detects a change, it dynamically updates the routes.

## Orchestrators

> While each provider is different, we can categorize them in four groups:

Reviewer: ```suggestion
While each provider is different, you can think of each as belonging to one of four categories:
```

- Label-based: each deployed container has a set of labels attached to it
- Key-Value-based: each deployed container updates a key-value store with relevant information
- Annotation-based: a separate object, with annotations, defines the characteristics of the container
- File-based: uses files to define configuration

## Provider Namespace

When you declare certain objects in the Traefik dynamic configuration,
such as middleware, services, TLS options or server transports, they reside in their provider's namespace.

> For example, if you declare a middleware using a Docker label, it resides in the docker provider namespace.

Reviewer: ```suggestion
For example, if you declare a middleware using a Docker label, it resides in the Docker provider namespace.
```

!!! important "Kubernetes Namespace"

    As Kubernetes also has its own notion of namespace,

>     one should not confuse the "provider namespace" with the "Kubernetes Namespace" of a resource when in the context of cross-provider usage.

Reviewer: ```suggestion
    one should not confuse the _provider namespace_ with the _Kubernetes Namespace_ of a resource when in the context of cross-provider usage.
```

    In this case, since the definition of a Traefik dynamic configuration object is not in Kubernetes,

>     specifying a "Kubernetes Namespace" when referring to the resource does not make any sense.

Reviewer: ```suggestion
    specifying a Kubernetes Namespace when referring to the resource does not make any sense.
```
